receive.c

```diff
@@ -121,10 +151,37 @@ static void on_read(int fd, char *buf, ssize_t len, char **buf1, ssize_t *len1)
 			} else {
 				struct JsonNode *json = json_decode(*buf1);
 				if(json != NULL) {
-					char *out = json_stringify(json, "\t");
-					printf("%s\n", out);
+					if(filtertype != 0) {
```

Comment:
What about turning this stuff in a seperate function? That would deduplicate the whole even more.